nevergrad/functions/images/imagelosses.py

```diff
@@ -1,35 +1,110 @@
+import cv2  # type: ignore
+import imquality.brisque as brisque  # type: ignore
+import lpips  # type: ignore
 import os
+import torch
 import typing as tp
 import numpy as np
 from nevergrad.functions.base import UnsupportedExperiment
+from nevergrad.common.decorators import Registry
+
+
+registry: Registry[tp.Any] = Registry()
 
 
 class ImageLoss:
     def __init__(self, reference: tp.Optional[np.ndarray] = None) -> None:
         pass
 
     def __call__(self, img: np.ndarray) -> float:
-        raise NotImplementedError
+        raise NotImplementedError(f"__call__ undefined in class {type(self)}")
 
 
-class SumAbsoluteDifferences(ImageLoss):
-    def __init__(self, reference: np.ndarray) -> None:
+class ImageLossWithReference(ImageLoss):
```

Comment:
i'm still uneasy with keeping these two classes for not much, another option would be to include this code in `ImageLoss` under a condition `self.REQUIRES_REFERENCE` (a class attribute, which defaults to True since there are fewer cases which do not need it)

Not sure it is simpler though. I probably just do not like the name at all, `ReferenceImageLoss` would be more appropriate for a class name in my opinion

Reply:
Done in the PR on PR.